### PR TITLE
Nj add other missing JS tests

### PR DESCRIPTION
**Pull request by cohesivejones**
__Description of Change(s) Introduced:__ 
Added missing tests for selectors, utilities and reducers

__Relevant github Issue:__ 
[#3]

### CODE REVIEW

File: frontend/reducers/test_reducer.test.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      )
+    ).toEqual({
+      'a-test-id': { id: 'a-test-id' }
+    });
+  });
+
+  it('merges test on RECEIVE_TEST', () => {
+    expect(
+      reducer(
+        {},
```
Comment (kamok) [inline frontend/reducers/test_reducer.test.js]: Recommend adding an actual Test instead of empty Object

File: frontend/util/test_api_util.test.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  });
+
+  describe('postTest', () => {
+    const response = {data: 'created-test-data'};
+    beforeEach(() => {
+      jest.spyOn(axios, 'post').mockImplementation(() => response);
+    });
+
+    it('calls post "api/tests"', async () => {
+      const test = {
```
Comment (kamok) [inline frontend/util/test_api_util.test.js]: Recommend using Faker for JS to improve readability and long term maintainability

Comment (kamok) [reply frontend/util/test_api_util.test.js]: https://github.com/marak/Faker.js/

Comment (kamok) [reply frontend/util/test_api_util.test.js]: https://github.com/rosiejs/rosie

File: frontend/util/test_submission_helper.test.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
+import * as TestSubmissionHelper from './test_submission_helper';
+
+describe('test_submission_helper', () => {
+  describe('isValidDueDate', () => {
+    it('returns true when a date with format YYYY-MM-DD is provided', () => {
```
Comment (kamok) [inline frontend/util/test_submission_helper.test.js]: Recommend using built-in jest test looping: https://jestjs.io/docs/en/api#1-testeachtable-name-fn-timeout to improve readability and be DRY
